app/subscribers/solidus_stripe/webhook/payment_intent_subscriber.rb

```diff
@@ -74,6 +80,36 @@ def extract_payment_from_event(event)
         payment_intent_id = event.data.object.id
         Spree::Payment.find_by!(response_code: payment_intent_id)
       end
+
+      def perform_full_capture(payment)
+        payment.complete!.tap do
+          SolidusStripe::LogEntries.payment_log(
+            payment,
+            success: true,
+            message: "Capture was successful after payment_intent.succeeded webhook"
+          )
+        end
+      end
+
+      def perform_partial_capture(payment, amount)
+        previous_money = payment.money
+        payment.amount = amount
+        payment.complete!.tap do
+          SolidusStripe::LogEntries.payment_log(
+            payment,
+            success: true,
+            message: <<~MSG
+              Partial capture (#{payment.money} of #{previous_money}) was successful after payment_intent.succeeded webhook. Payment amount was updated
+            MSG
+          )
+        end
+      end
```

Comment:
I'm not 100% sure about this part 🤔 
The fact is that even if we have a partial capture, in this scenario, we also have a partial refund.
There is a `charge.captured` event just before the `payment_intent.succeeded` with the following params:
- "amount"
- "amount_captured"
- "amount_refunded"

But even considering the `amount_received` from `payment_intent.succeeded` (which I think is also right) my question is:
Would it be better if we create a refund using the `amount_refunded` value (or the `amount - amount_received`) rather than updating the payment amount? What do you think?

Reply:
Hmm. I see your point. The thing is that we don't get a `charge.refunded` webhook event, although Stripe does show a "Partial refund" label. At first sight, I don't see an associated refund record, though. I'll study it further. I'll put the PR as WIP in the meantime. Thanks for your feedback.

Reply:
@rainerdema, I confirm that even if the Stripe UI and the received webhooks differ, a Stripe Refund object is created under the hood for partial captures. Therefore, it makes total sense also to create if on our end. Thanks for pointing that out. I'm going to work on that straightaway.